### PR TITLE
healthCheckImprovement[transitTimeout]

### PR DESCRIPTION
**Pull request by xinlian12**
**Issue:**
One of the main mechanism within SDK to detect a broken connection is during the health check flow. The current implementation will not be able to detect a broken connection under relative high workload.

**Soltuion**
when a connection is in a broken status - usually it will cause the network request fail with transit timeout (client side generated timeout). In this PR, we are adding a tracking for continuous transitTimeout counter, and integrate it into the healthy check flow to help detect the broken connection faster.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java [hunk 1]
```diff
@@ -474,6 +474,8 @@ public static final class Options {
         @JsonProperty()
         private final Duration sslHandshakeTimeoutMinDuration;
 
+        @JsonProperty()
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java:477]: should we add this to the Options builder example / java doc that we have?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java [hunk 2]
```diff
@@ -541,6 +544,7 @@ private Options(final ConnectionPolicy connectionPolicy) {
             this.tcpKeepIntvl = 1; // Configuration for EpollChannelOption.TCP_KEEPINTVL
             this.tcpKeepIdle = 30; // Configuration for EpollChannelOption.TCP_KEEPIDLE
             this.sslHandshakeTimeoutMinDuration = Duration.ofSeconds(5);
+            this.transientTimeoutDetectionThreshold = 3;
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java:547]: may be add some java docs explaining internally how this works. or even comments.

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @xinlian12 , great work!

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java (hunk 1 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java:478]: added javadoc

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java (hunk 2 above)
Comment (xinlian12) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java:548]: added more comments

**Pull request merged**